airflow/www/views.py

```diff
@@ -4359,10 +4359,7 @@ def process_form(self, form, is_created):
                 conn_type_from_extra_field = key.split("__")[1]
                 if conn_type_from_extra_field == conn_type:
                     value = form.data[key]
-                    # Some extra fields have a default value of False so we need to explicitly check the
-                    # value isn't an empty string.
-                    if value != "":
-                        extra[field_name] = value
+                    extra[field_name] = value
```

Comment:
Hmm. The original comment explicitly mentioned case with default = False - has it been checked? I think we should find a "default False" field and test it - and my feeling is that somewhere we should keep the list of fields that have default = False and handle them separately.

Reply:
I have checked and in the current implementation, which we have now in Airflow, all False sent to the server, but as I understood from this comment they shouldn't. I don't understand what was the reason for this if-clause, because if we leave the check for `""` then users can't remove their values from the field in Connection form or if we change to the `False` then users can't unchecked their check boxes in the form.
@josh-fell could you please explain what behavior is expected for the form validation with `""` if-clause?

Reply:
Hmm. I think it's nicely described in https://github.com/apache/airflow/pull/23241 why the if was added

Reply:
Basically it is to avoid keys from "other" connection types to be added when you add a connection. That's basically a consequence of the (bad) way the form is implemented - where basically the form contains all possible fields from all connection types registered in the system - but they are dynamically shown/hidden based on connection type - so that DOM does not have to be dynamically updated based on connection type besides showing/hiding.

Reply:
@potiuk hello, maybe we can change code something like this:
```
if value != "":
    extra[field_name] = value
elif field_name in extra:
    del extra[field_name]
```
in this case we still save previous behavior and, also, we can delete value if the user decide delete it on UI.

Reply:
Might be, I am still not sure what exactly scenario was described in https://github.com/apache/airflow/pull/23241 - but maybe (again) @dstandish and @josh-fell can shed some light on it.

Reply:
Oddly enough @dstandish [asked the same question](https://github.com/apache/airflow/pull/23241#issuecomment-1109814135) in the original PR. Effectively what was happening is non-string fields like those which were declared as `BooleanField` _or_ `IntegerField`s were being applied behind the scenes. Therefore, generating a connection URI was incorrect and included an attribute that was completely unrelated to another connection in the majority of cases.

Reply:
Small edit: ...which were declared as `BooleanField` or `IntegerField`s _with a default value_ were being applied. The default value is key here. This is what the intention of the PR was attempting to prevent.

> >
> 
> @potiuk hello, maybe we can change code something like this:
> 
> if value != "":
>     extra[field_name] = value
> elif field_name in extra:
>     del extra[field_name]
> 
> in this case we still save previous behavior and, also, we can delete value if the user decide delete it on UI.
> 

@MaksYermak this alternative you proposed seems reasonable to me!

Reply:
I think yes. I have updated the code with the last push

Reply:
Hi @potiuk ! Can you please check if we still can merge this logic? Thanks!